roles/promtail/tasks/deploy.yml

```diff
@@ -58,7 +58,7 @@
     owner: "promtail"
     group: "root"
     mode: "0644"
-    validate: "/usr/bin/promtail -check-syntax -config.file %s"
+    validate: "/usr/bin/promtail -check-syntax -config.file %s {{ promtail_extra_args }}"
```

Comment:
```suggestion
    validate: "/usr/bin/promtail -check-syntax -config.file %s"
```

Reply:
This step seemingly fails when templating with environment variables without also passing in the additional flags: `invalid character \"{\" in host name. Use '-config.expand-env=true' flag if you want to expand environment variables in your config file`

So I believe I will also need to pass those flags into that command as well as the systemd unit?

Reply:
Applogies, I am not super familiar with Ansible but would something along these lines be reasonable? Presumably I would also need to add some check for the list length being >0 as to not add those random spaces?

```yaml
validate: "/usr/bin/promtail {{ promtail_extra_args | join(' ') }} -check-syntax -config.file %s"
```

Reply:
Could you please share exactly your implementation? Where you apply env and how you are using it.

Reply:
I believe the issue is with your implementation, not with Ansible. Validation **not** **require** additional flags.

Reply:
I don't think this is to do with Ansible but with how Promtail runs with the `-check-syntax` flag?
```bash
~/tmp
❯ cat config.yaml
client:
  url: https://${loki_base_url}/loki/api/v1/push
~/tmp
❯ promtail -check-syntax -config.file config.yaml
Unable to parse config: config.yaml: parse "https://${loki_base_url}/loki/api/v1/push": invalid character "{" in host name. Use `-config.expand-env=true` flag if you want to expand environment variables in your config file
~/tmp
❯ promtail -check-syntax -config.expand-env=true -config.file config.yaml
Valid config file! No syntax issues found
```

Reply:
Got it. I’m unclear why you’re configuring clients with environment variables when you could directly set the variable in the inventory or use facts to populate the variable content.

Please adjust it as follows:

```yaml
validate: "/usr/bin/promtail -config.expand-env=true -check-syntax -config.file %s"
```

We should not include additional flags. Add only `-config.expand-env=true`.

Once this adjustment is made, I will approve the change.

Reply:
I might have a slightly different use-case for this role to most but I am using this with Packer to build an image which will be used across different environments so trying to avoid hardcoding any URLs into the image itself.

I have made the adjustment, thanks for your suggestions and being so responsive.